setuptools/extension.py

```diff
@@ -27,8 +27,10 @@ def _have_cython():
 # for compatibility
 have_pyrex = _have_cython
 if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
     # Work around a mypy issue where type[T] can't be used as a base: https://github.com/python/mypy/issues/10962
-    _Extension = distutils.core.Extension
+    _Extension: TypeAlias = distutils.core.Extension
```

Comment:
```suggestion
    from distutils.core import Extension as _Extension
```

And then we can drop the `TypeAlias`.